conformance/lang/expressions/new-expr/explicit-new-expr/stream_class_descriptor.balt

```diff
@@ -0,0 +1,167 @@
+Labels: new-expr, stream-type-descriptor, int-type-descriptor
+
+type IntRecord record {|
+    int value;
+|};
+
+function getRecordValue(IntRecord|error? returnedVal) returns int|error? {
+    if (returnedVal is IntRecord) {
+        return returnedVal.value;
+    } else {
+        return returnedVal;
+    }
+}
+
+function init() {
+    stream<int, error?> emptyStream = new stream<int, error?>();
+    int|error? d = getRecordValue(emptyStream.next());
+    io:println(d); // @output
+}
+
+Test-Case: output
+Description: Test new-expr to create a stream without completion type
+Labels: new-expr, stream-type-descriptor
+
+type IntRecord record {|
+    int value;
+|};
+
+function getRecordValue(IntRecord? returnedVal) returns int? {
+    if (returnedVal is IntRecord) {
+        return returnedVal.value;
+    } else {
+        return returnedVal;
+    }
+}
+
+function init() {
+    stream<int> emptyStream = new stream<int>();
+    int? d = getRecordValue(emptyStream.next());
+    io:println(d); // @output
+}
+
+Test-Case: output
+Description: Test new-expression with int streams.
+Labels: new-expr, stream-type-descriptor, int-type-descriptor
+
+type IntRecord record {|
+    int value;
+|};
+
+function getRecordValue(IntRecord|error? returnedVal) returns int|error? {
+    if (returnedVal is IntRecord) {
+        return returnedVal.value;
+    } else {
+        return returnedVal;
+    }
+}
+
+class NumbersGenerator {
+    int i = 0;
+    public isolated function next() returns IntRecord|error? {
+        self.i += 1;
+        return { value: self.i };
+    }
+}
+
+function init() {
+    NumbersGenerator numbersGenerator = new;
+    stream<int, error?> numberStream = new stream<int, error?>(numbersGenerator);
+    int|error? a = getRecordValue(numberStream.next());
+    int|error? b = getRecordValue(numberStream.next());
+    io:println(a); // @output 1
+    io:println(b); // @output 2
+}
+
+Test-Case: output
+Description: Test closing of stream created with new-expression.
+Labels: new-expr, stream-type-descriptor, int-type-descriptor
+
+type IntRecord record {|
+    int value;
+|};
+
+function getRecordValue(IntRecord|error? returnedVal) returns int|error? {
+    if (returnedVal is IntRecord) {
+        return returnedVal.value;
+    } else {
+        return returnedVal;
+    }
+}
+
+class NumbersGenerator {
+    int i = 0;
+    public isolated function next() returns IntRecord|error? {
+        self.i += 1;
+        return { value: self.i };
+    }
+}
+
+function init() {
+    NumbersGenerator numbersGenerator = new;
+    stream<int, error?> numberStream = new stream<int, error?>(numbersGenerator);
+    int|error? a = getRecordValue(numberStream.next());
+    int|error? b = getRecordValue(numberStream.next());
+    io:println(a); // @output 1
+    io:println(b); // @output 2
+    error? closed = numberStream.close();
+    io:println(closed is ()); // @output true
+}
+
+Test-Case: panic
+Description: Test closed stream next.
```

Comment:
Shall we check the auto-closing behavior as well?

Reply:
for example, calling, `next` after `next` returns (), should panic

Reply:
I am waiting on https://github.com/ballerina-platform/ballerina-spec/issues/1036

Reply:
Spec issue is updated with details

Reply:
Tracked by https://github.com/ballerina-platform/ballerina-lang/issues/34934